src/ResourceTile/ResourceTile.tsx

```diff
@@ -36,96 +30,82 @@ export const CssClasses = {
-  [IconOrientation.LANDSCAPE]: "ResourceTile--icon--landscape",
-  [IconOrientation.PORTRAIT]: "ResourceTile--icon--portrait",
-  [IconOrientation.UNKNOWN]: "",
+const IconOrientationCssClasses: Record<IconOrientation, string> = {
+  landscape: "ResourceTile--icon--landscape",
+  portrait: "ResourceTile--icon--portrait",
+  unknown: "",
 };
 
-interface BasicProps {
-  icon: string;
-  title: string;
-  url: string;
+export interface Action {
+  content: React.ReactNode;
+  href?: string;
+  onClick?: () => void;
 }
 
-/**
- * These props may be passed through several component layers, for example:
- * ResourceLinkOnDistrictPage --> AppLinkOnDistrictPage --> AppLink --> ResourceLink
- */
-export interface PassThroughProps {
-  actions?: { content: React.ReactNode; href?: string; onClick?: () => void }[];
+export interface Props {
+  actions: [] | [Action] | [Action, Action];
   additionalInfo?: React.ReactNode;
   className?: string;
   draggable?: boolean;
+  icon: string;
+  loadingSpinner: React.ReactNode;
   notes?: string;
   notify?: boolean;
   onClick?: (event?: React.MouseEvent) => void;
+  openInSameTab: boolean;
   overlays?: React.ReactNode[];
-  size?: IconSize;
-}
-
-export function extractPassThroughProps(props: any): PassThroughProps {
-  return _.pick(props, [
-    "actions",
-    "additionalInfo",
-    "draggable",
-    "className",
-    "notes",
-    "notify",
-    "onClick",
-    "overlays",
-    "size",
-  ]);
+  size: IconSize;
+  title: string;
+  url: string;
 }
 
-type Props = BasicProps & PassThroughProps;
-
 interface State {
   iconOrientation: IconOrientation;
 }
 
 export class ResourceTile extends React.PureComponent<Props, State> {
-  static defaultProps = {
+  static defaultProps: Pick<Props, "actions" | "size" | "openInSameTab"> = {
     actions: [],
-    size: IconSize.LARGE,
+    size: "large",
+    openInSameTab: false,
   };
 
-  iconRef;
+  iconRef: React.RefObject<HTMLImageElement>;
 
-  constructor(props) {
+  constructor(props: Props) {
     super(props);
 
     this.iconRef = React.createRef();
     this.state = {
-      iconOrientation: IconOrientation.UNKNOWN,
+      iconOrientation: "unknown",
     };
   }
 
   determineIconOrientation() {
     const element = this.iconRef.current;
 
-    let iconOrientation = IconOrientation.UNKNOWN;
+    let iconOrientation: IconOrientation = "unknown";
     if (element) {
       const height = element.height;
       const width = element.width;
-      iconOrientation = height > width ? IconOrientation.PORTRAIT : IconOrientation.LANDSCAPE;
+      iconOrientation = height > width ? "portrait" : "landscape";
     }
     this.setState({ iconOrientation });
   }
 
   renderIcon() {
-    const { icon, title } = this.props;
+    const { icon, title, loadingSpinner } = this.props;
     const { iconOrientation } = this.state;
     const loading = icon === "";
 
     if (loading) {
-      return <LoadingSpinner size={LoadingSpinner.Size.S} />;
+      return loadingSpinner;
```

Comment:
I'm open to thoughts on if this makes sense - previously, if the icon is an empty string we displayed a `<LoadingSpinner>`. However, we actually don't have a `LoadingSpinner` component in Dewey, each repo has its own version. As a way around this, we require the user of this component to pass in a `loadingSpinner` prop to be used in this case. It does feel a little weird that we have this special case that comes up when icon is an empty string. Maybe the prop types should make icon and loadingSpinner mutually exclusive and we use whichever is given?

Reply:
Mmm interesting. Feels like the real generalization here is to allow consumers to pass in a `ReactNode` for the icon, where that node isn't necessarily a loading spinner. What if we make a new optional `iconNode` prop for this? Open to other names!